File: HackerRank/30 Days of Code/Day_10_Binary_Numbers.cpp
/*    ------------------------------------------------------------------------------------------
 #    Author    : Mohammad Montasim -Al- Mamun Shuvo
 #    Copyright : Copyright 2020, Mohammad Montasim -Al- Mamun Shuvo
 #    Email     : montasimmamun@gmail.com
 #    Github    : https://github.com/montasimmamun/

 #    Date      : Created on 18/07/2020
 #    Version   : 1.0.1
 #    ------------------------------------------------------------------------------------------*/

#include <iostream>
using namespace std;
int main()
{
    int array[10], input, count = 0, i;

    cin >> input;

    for (i = 0; input > 0; i++)
    {
        array[i] = input % 2;
        input = input / 2;
    }

    for (i = i - 1; i >= 0; i--)
    {
        if (array[i] == 1)
        {
            if (array[i] == 1 && array[i + 1] == 1)
            {
                count += 2;
            }
            else
            {
                count = 1;
            }
        }
    }

    cout << count;
}

File: HackerRank/30 Days of Code/Day_10_Binary_Numbers.py
#    /* ------------------------------------------------------------------------------------------
#    Author    : Mohammad Montasim -Al- Mamun Shuvo
#    Copyright : Copyright 2020, Mohammad Montasim -Al- Mamun Shuvo
#    Email     : montasimmamun@gmail.com
#    Github    : https://github.com/montasimmamun/

#    Date      : Created on 21/07/2020
#    Version   : 1.0.1
#    ------------------------------------------------------------------------------------------*/


import math
import os
import random
import re
import sys


if __name__ == '__main__':
    num = int(input())

    result = 0
    maximum = 0

    while num > 0:
        if num % 2 == 1:
            result += 1
            if result > maximum:
                maximum = result

        else:
            result = 0

        num //= 2

    print(maximum)
